src/plugins/saved_objects_management/public/register_services.ts

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { StartServicesAccessor } from '../../../core/public';
+import { createSavedSearchesLoader } from '../../discover/public';
+import { SavedObjectsManagementPluginStart, StartDependencies } from './plugin';
+import { ISavedObjectsManagementServiceRegistry } from './services';
+
+export const registerServices = async (
+  registry: ISavedObjectsManagementServiceRegistry,
+  getStartServices: StartServicesAccessor<StartDependencies, SavedObjectsManagementPluginStart>
+) => {
+  const [coreStart, { dashboard, data, visualizations }] = await getStartServices();
+
+  if (dashboard) {
+    registry.register({
+      id: 'savedDashboards',
+      title: 'dashboards',
+      service: dashboard.getSavedDashboardLoader(),
+    });
+  }
+
+  if (visualizations) {
+    registry.register({
+      id: 'savedVisualizations',
+      title: 'visualizations',
+      service: visualizations.savedVisualizationsLoader,
+    });
+  }
```

Comment:
So. In theory, this is bad IOC. These registrations should be performed from the `dashboard` and `visualizations` plugins, using the SOM plugin APIs. However, as one of the point of https://github.com/elastic/kibana/issues/62048 is to try to remove these `loaders` aka `services`, I reversed the dependencies and kept the registration in the SOM plugin instead, to facilitate the future removal and to avoid polluting the plugins. 

For the record, this was previously done in `src/legacy/core_plugins/kibana/public/management/saved_object_registry.ts `